python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -211,14 +212,30 @@ def test_NDCG(self):
         # Expected result: 1.0 (perfect ranking)
         assert result == 1.0
 
-    def test_NDCG_mask(self):
+    def test_NDCG_2(self):
+        # Create an instance of the NDCG metric
+        ndcg_metric = NDCG()
+
+        # Define the true relevance scores and predicted scores
+        y_true = tf.constant([[3.0, 2.0, 1.0], [1.0, 2.0, 3.0]])
+        y_pred = tf.constant([[0.3, 0.2, 0.1], [30, 20, 10]], dtype=tf.float32)
+
+        # Update the metric state
+        ndcg_metric.update_state(y_true, y_pred)
+
+        # Retrieve the result
+        result = ndcg_metric.result().numpy()
+
+        assert np.isclose(result,  0.894999, atol=0.0001)
```

Comment:
there is a lot or replication between `test_NDCG_1` and `test_NDCG_2`. Typically, the name should point to what is the difference. 

One thing we can do to improve such tests is to parametrize with pytest. Pytest.parametrize does not work well with unittest classes but here we don't get something subclassing `RankingMetricsTest` from unittest. So we could: 
```python
    @pytest.mark.parametrize("y_true,expected_result", [[10, 20, 30],[30, 20, 10]],
                                                   [1.,, 0.895])
    def test_NDCG(self, y_true, expected_result):
        .... 
```

Reply:
done

Reply:
Did you check this in? I still see the old code @mohazahran

Reply:
ping

Reply:
Yes I removed the code duplication.